### DISCUSSION TITLE
Objects List unavailable. Please review your WebSockets configuration and try again

### DISCUSSION THREAD

**Discussion by 1934801322** [minio/minio#17371] [Q&A] [answered]
## Expected Behavior
I want the data in my bucket to be displayed and downloaded normally

## Current Behavior
When I use the web server to proxy the past to the minio service bucket, the web interface will pop up an error
my error message：Couldn't establish WebSocket connection. Please review your configuration and try again.
The error message of my bucket list is that there is no data in the bucket：Objects List unavailable. Please review your WebSockets configuration and try again

## Possible Solution
I tried to modify the configuration of the web proxy server
This is the configuration information of my nginx proxy server
    location / {
        proxy_pass http://192.168.0.4:9000;
            # To support websocket
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
            
           
    }

## Your Environment
My Linux server is Centos07
My minio version is
minio -version
minio version RELEASE.2023-06-02T23-17-26Z (commit-id=0649aca2195a8d3aef9b946337b4ca6581e23844)
Runtime: go1.19.9 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.

I would be very grateful if you could help me

**Comment by klauspost** [comment]
It appears you set it for the S3 API port, but not the console port (probably 9001). Either way consult the documentation or search here on how others fixed the issue.

**Reply by 1934801322** [reply]
Thank you for your answer. I just checked and found out that my configuration is not comprehensive enough

**Comment by luodl1991** [comment]
你好  请问这个问题解决了吗？ 我也遇到了相同的问题。

已解决：
    location / {
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-NginX-Proxy true;

        # This is necessary to pass the correct IP to be hashed
        real_ip_header X-Real-IP;

        proxy_connect_timeout 300;

        # To support websocket
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";

        chunked_transfer_encoding off;

        proxy_pass http://51pla_img_minio/;
    }

**Reply by 1934801322** [reply, marked as answer]
你好！我这个问题解决了。我目前的location 配置是这样的。你可以根据以下配置参考
  location / {
   #
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;

    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";

    proxy_connect_timeout 300;
    proxy_read_timeout 300;
    proxy_pass http://192.168.0.4:9000/; # This uses the upstream directive definition to load balance   }
}

**Reply by luodl1991** [reply]
感谢你的回复，已解决当前问题。

**Comment by ShojibFlamon** [comment]
# Minio server configuration
# It works fine for me

```
server {
listen 80;
listen [::]:80;

    server_name minio.local;
    # Don't fogot to add hosts file entry

    location / {

        proxy_pass http://11.1.2.2:9001;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        
        # To support websocket
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }

    error_log /var/log/nginx/minio.error.log;
    access_log /var/log/nginx/minio.access.log;
}
```
